internal/guest/runtime/runc/container.go

```diff
@@ -389,6 +400,70 @@ func (c *container) startProcess(
 		}
 	}
 
+	// This is to enabling container logging in Bare Metal Containers. This is in experimental stage now and need to be revisited.
+	var stdoutFifoPipe, stderrFifoPipe *os.File
+	if annotations != nil {
+		pipeNameSuffix, exists := annotations["io.microsoft.bmc.logging.pipelocation"]
+		pipeDirectory := "/run/gcs/containerlogs/"
+		stdoutPipeName := pipeDirectory + pipeNameSuffix + "-stdout"
+		stderrPipeName := pipeDirectory + pipeNameSuffix + "-stderr"
+		os.MkdirAll(pipeDirectory, os.ModePerm)
+		if exists {
+			_, err = os.Stat(stdoutPipeName)
+			if err != nil {
+				// fifo pipe does not exist, create one
+				err = syscall.Mkfifo(stdoutPipeName, 0666)
+				if err != nil {
+					msg := "Error creating named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+					logrus.Infof(msg)
+					return nil, errors.Wrapf(err, msg)
+				}
+			}
+
+			_, err = os.Stat(stderrPipeName)
+			if err != nil {
+				// fifo pipe does not exist, create one
+				err = syscall.Mkfifo(stderrPipeName, 0666)
+				if err != nil {
+					msg := "Error creating named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+					logrus.Infof(msg)
+					return nil, errors.Wrapf(err, msg)
+				}
+			}
+
+			// pipe either exist before hand or we have created one above
+			stdoutFifoPipe, err = os.OpenFile(stdoutPipeName, os.O_RDWR|os.O_APPEND, os.ModeNamedPipe)
+			if err != nil {
+				msg := "Error opening named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+				logrus.Infof(msg)
+				return nil, errors.Wrapf(err, msg)
+			}
+
+			stderrFifoPipe, err = os.OpenFile(stderrPipeName, os.O_RDWR|os.O_APPEND, os.ModeNamedPipe)
+			if err != nil {
+				msg := "Error opening named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+				logrus.Infof(msg)
+				return nil, errors.Wrapf(err, msg)
+			}
+		}
+
+		isLoggingSideCarContainerStr, exists := annotations["io.microsoft.bmc.logging.isLoggingSideCarContainer"]
+		if exists {
+			isLoggingSideCarContainer, _ := strconv.ParseBool(isLoggingSideCarContainerStr)
```

Comment:
Why are we not returning the error here? If someone has a bad value here (say something like "truee") that's meant to indicate "true", when the parse fails, we'd get a `false` value for `isLoggingSideCarContainer`, which seems like it could be an issue.